pre_install_report/library/pre_install_check.py

```diff
@@ -91,6 +92,24 @@ def _parse_release_info(self, release_info):
             print(viya_messages.KUBELET_VERSION_ERROR)
             sys.exit(viya_messages.BAD_OPT_RC_)
 
+    def _get_server_git_version(self, utils):
+        """
+        Retrieve the Kubernetes server version and validate the git version
+        """
+        try:
+            versions: Dict = utils.get_k8s_version()
+            server_version = versions.get('serverVersion')
+            git_version = server_version.get('gitVersion')
+            self.logger.info("git_version {} ".format(str(git_version)))
+
+            if git_version.startswith("v"):
+                git_version = git_version[1:]
+            self.logger.debug('Server kubectl version = {}'.format(git_version))
+            return git_version
```

Comment:
Would it be worth adding a check to make sure the version is in an expected format, or at least not empty? Not sure if there could be a scenario where things aren't up enough or the user doesn't have permissions and an empty value is returned.

Reply:
I looked at the Semantic Version https://semver.org/ supported by kubernetes https://kubernetes.io/releases/
I will add following checks for the major and minor version which are relevant for comparison as follows:

**A normal version number MUST take the form X.Y.Z where X, Y, and Z are non-negative integers, and MUST NOT contain leading zeroes.** X is the major version, Y is the minor version, and Z is the patch version. Each element MUST increase numerically. For instance: 1.9.0 -> 1.10.0 -> 1.11.0.

Good catch!

Reply:
Code Updated.

Reply:
Updated code to check version formatting.  Completed.